### PR TITLE
-deprecate ContentCard class

### PR DESCRIPTION
**Pull request by sbenedicadb**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Documentation/README.md [hunk 1]
```diff
@@ -36,7 +36,7 @@
 - Developer Documentation
   - [API usage](./sources/api-usage.md#Message Feed and Code-based experiences APIs)
```
Comment (PravinPK) [inline Documentation/README.md:37]: Formatting needs fixing here and we can remove `Feed` and state Content Card

Screenshot : 
Screenshot 2024-08-19 at 8 01 47 AM

**Review by PravinPK — COMMENTED** [review]
Should we also edit the statement in the FeedItem class class here
https://github.com/adobe/aepsdk-messaging-android/blob/4b7584791ee06814ed3ce98ebf9c63bc451fde5e/code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/FeedItem.java#L23

We can recommend to use ContentCardSchemaData Instead

File: Documentation/README.md (hunk 1 above)
Comment (sbenedicadb) [reply Documentation/README.md]: good catch - there are 3 on this page that are broken - fixed them all

File: code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCard.java [hunk 2]
```diff
@@ -22,7 +22,10 @@
  * A {@link ContentCard} object encapsulates the information necessary for a non-disruptive yet
  * interactive offer. Customers can use the Messaging SDK to render the content card in a
  * pre-defined format or implement their own rendering.
+ *
+ * @deprecated Use {@link ContentCardSchemaData} instead
```
Comment (spoorthipujariadobe) [inline code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCard.java:26]: Can we make the same change in `FeedItem.java` as well?

**Review by spoorthipujariadobe — COMMENTED** [review]
You can cherry pick this commit to fix failing unit tests
https://github.com/adobe/aepsdk-messaging-android/pull/263/commits/8f9a7d61238984b80eb8c637fbf06f9b5c667210

File: code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCard.java (hunk 2 above)
Comment (sbenedicadb) [reply code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCard.java:26]: done

**Review by spoorthipujariadobe — APPROVED** [review]

**Pull request merged**